java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -199,6 +200,9 @@ private void setValueFromParser(ValueVector valueVector, int i) throws IOExcepti
     case FLOAT8:
       ((Float8Vector)valueVector).getMutator().set(i, parser.readValueAs(Double.class));
       break;
+    case VARBINARY:
+      ((VarBinaryVector)valueVector).getMutator().setSafe(i, parser.readValueAs(String.class).getBytes(UTF_8));
+      break;
```

Comment:
you can use commons-codec's Hex class:
you can add this dependency to the pom:
```
    <dependency>
      <groupId>commons-codec</groupId>
      <artifactId>commons-codec</artifactId>
      <version>1.10</version>
      <scope>compile</scope>
    </dependency>
```

If you don't want to add a dependency you can also convert one byte at a time:
String.format("%02x", byteValue);